### PR TITLE
Changes for Android build of Foundation

### PR DESCRIPTION
**Pull request by johnno1962**
Minor changes to build-script-imll to support build of Foundation with Android toolchain (see https://github.com/apple/swift-corelibs-foundation/pull/622) Simply passes additional Android specific options though to ninja script builder. Has been tested with Linux and OSX builds and doesn’t seem to break anything.

### CODE REVIEW

File: utils/build-script-impl [hunk 1]
```diff
@@ -2259,7 +2259,17 @@ for host in "${ALL_HOSTS[@]}"; do
...
-                if [[ $(is_cross_tools_host ${host}) ]]; then
+                FOUNDATION_CFLAGS="CFLAGS=${CFLAGS}"
+                FOUNDATION_SFLAGS="SWIFTCFLAGS=${SWIFTCFLAGS}"
+                FOUNDATION_LDFLAGS="LDFLAGS=${LDFLAGS}"
+                if [[ "${ANDROID_NDK}" != "" ]]; then
+                    FOUNDATION_CFLAGS="CFLAGS=-DDEPLOYMENT_TARGET_ANDROID --sysroot=${ANDROID_NDK}/platforms/android-${ANDROID_API_LEVEL}/arch-arm -I${ANDROID_ICU_UC}/include -I\${SDKROOT}/lib/swift"
+                    FOUNDATION_SFLAGS="SWIFTCFLAGS=-DDEPLOYMENT_TARGET_ANDROID -I${ANDROID_NDK}/platforms/android-${ANDROID_API_LEVEL}/arch-arm/usr/include"
+                    FOUNDATION_LDFLAGS="LDFLAGS=-fuse-ld=gold --sysroot=${ANDROID_NDK}/platforms/android-21/arch-arm -L${ANDROID_NDK}/toolchains/arm-linux-androideabi-4.9/prebuilt/linux-x86_64/lib/gcc/arm-linux-androideabi/4.9.x -L${ANDROID_ICU_UC} -L\${SDKROOT}/lib/swift/android -ldispatch"
+                    FOUNDATION_BUILD_ARGS+=(
+                        "--target=armv7-none-linux-androideabi"
```
Comment (karwa) [inline utils/build-script-impl]: Why isn't this set below? Is android not a cross-compilation target?

**Comment by karwa** [comment]
I'm not sure this is entirely correct. Foundation is a target library but currently the build script only builds for the host. We only ever cross-compile for Android.

Would it be possible to wait until we have restructured the build-script in to what @ddunbar calls a toolchain-based process? I hope to have a patch to do that soon.

**Comment by johnno1962** [comment]
Sure, This change can keep and admittedly it is not a tidy one. I’m only keen for the swift-corelibs-foundation changes to be merged before they "go stale"

**Comment by slavapestov** [comment]
@modocache Can you take a look?

**Comment by johnno1962** [comment]
@slavapestov @modocache. This PR has probably been superceeded by work to bring the Android build into the fold @gonzalolarralde & others are doing at the moment. I should have closed this a while ago.

**Pull request closed without merging**